opengrok-indexer/src/main/java/org/opengrok/indexer/web/Scripts.java

```diff
@@ -111,11 +111,14 @@ public String toHtml() {
         SCRIPTS.put("jquery-caret", new FileScript("js/jquery.caret-1.5.2.min.js", 25));
     }
 
+    private static final Comparator<Script> SCRIPTS_COMPARATOR = Comparator
+            .comparingInt(Script::getPriority)
+            .thenComparing(Script::getScriptData);
+
     /**
-     * Scripts which will be written to the page. We assume that the length
-     * could be the same as for {@link #SCRIPTS}.
+     * Scripts which will be written to the page.
      */
-    private final List<Script> outputScripts = new ArrayList<>(SCRIPTS.size());
+    private final NavigableSet<Script> outputScripts = new TreeSet<>(SCRIPTS_COMPARATOR);
```

Comment:
why exactly do you need NavigableSet ?

Reply:
to have it sorted based on the priority, because some scripts are dependent on each other